analyzer/protocol/stun/stun.evt

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2021 by the Zeek Project. See LICENSE for details.
+
+protocol analyzer spicy::STUN over UDP:
+	parse with STUN::STUNPacket;
+
+import STUN;
+
+on STUN::STUNPacket -> event STUN::STUNPacket($conn, $is_orig, self.M, self.C, self.trans_id);
+
+# SOFTWARE
+on STUN::Attribute::DESCRIPTION -> event STUN::string_attribute($conn, $is_orig, self.method,
```

Comment:
Can you explain why we trigger separate events for each attribute? This analyzer should only parse `STUNPacket` which already contain all the `Attribute`s.

Even if this currently doesn't e.g., produce double logging, I wonder if it wouldn't be simpler to make the handler for `STUN::STUNPacket` more powerful, so we should have a good reason for this complication.

Reply:
I don't understand what you mean.  Each type is different and has to be parsed differently depending on the attr type.

Reply:
We raise an event when we parsing any `Attribute`, but also when parsing a `STUNPacket`. `Attribute`s are always inside a `STUNPacket`. This means that for each `STUNPacket` (the top-level parsing unit) we raise an event for the full package, and then again for each of their `attributes`.

I was wondering whether we couldn't instead raise an event only for `STUNPacket` and do any Zeek-side, attribute-specific processing starting from their `attributes`.

Reply:
That would probably do the same thing.  I was a) reusing parts of the original code when I wrote this and this is how it was, and b) it just made sense in my head this way.  This would be quite a bit of code rewriting, if I'm understanding you correctly.